samples/snippets/create_custom_entry.py

```diff
@@ -0,0 +1,169 @@
+
+    # -------------------------------
+    # Currently, Data Catalog stores metadata in the
+    # us-central1 region.
+    # -------------------------------
+    location = "us-central1"
+
+    # -------------------------------
+    # Use Application Default Credentials to create a new
+    # Data Catalog client. GOOGLE_APPLICATION_CREDENTIALS
+    # environment variable must be set with the location
+    # of a service account key file.
+    # -------------------------------
+    datacatalog = datacatalog_v1.DataCatalogClient()
+
+    # -------------------------------
+    # 1. Environment cleanup: delete pre-existing data.
+    # -------------------------------
+    # Delete any pre-existing Entry with the same name
+    # that will be used in step 3.
+    expected_entry_name = datacatalog_v1.DataCatalogClient.entry_path(
+        project_id, location, entry_group_id, entry_id
+    )
+
+    try:
+        datacatalog.delete_entry(name=expected_entry_name)
+    except (NotFound, PermissionDenied):
+        pass
+
+    # Delete any pre-existing Entry Group with the same name
+    # that will be used in step 2.
+    expected_entry_group_name = datacatalog_v1.DataCatalogClient.entry_group_path(
+        project_id, location, entry_group_id
+    )
+
+    try:
+        datacatalog.delete_entry_group(name=expected_entry_group_name)
+    except (NotFound, PermissionDenied):
+        pass
+
+    # Delete any pre-existing Template with the same name
+    # that will be used in step 4.
+    expected_template_name = datacatalog_v1.DataCatalogClient.tag_template_path(
+        project_id, location, tag_template_id
+    )
+
+    try:
+        datacatalog.delete_tag_template(name=expected_template_name, force=True)
+    except (NotFound, PermissionDenied):
+        pass
+
+    # -------------------------------
+    # 2. Create an Entry Group.
+    # -------------------------------
+    entry_group_obj = datacatalog_v1.types.EntryGroup()
+    entry_group_obj.display_name = "My awesome Entry Group"
+    entry_group_obj.description = "This Entry Group represents an external system"
+
+    entry_group = datacatalog.create_entry_group(
+        parent=datacatalog_v1.DataCatalogClient.common_location_path(
+            project_id, location
+        ),
+        entry_group_id=entry_group_id,
+        entry_group=entry_group_obj,
+    )
+    print("Created entry group: {}".format(entry_group.name))
+
+    # -------------------------------
+    # 3. Create an Entry.
+    # -------------------------------
+    entry = datacatalog_v1.types.Entry()
+    entry.user_specified_system = "onprem_data_system"
+    entry.user_specified_type = "onprem_data_asset"
+    entry.display_name = "My awesome data asset"
+    entry.description = "This data asset is managed by an external system."
+    entry.linked_resource = "//my-onprem-server.com/dataAssets/my-awesome-data-asset"
+
+    # Create the Schema, this is optional.x
+    columns = []
```

Comment:
Rather than create `columns = []` and extend `entry.schema.columns`, I think appending directly to `entry.schema.columns` is supported.

https://proto-plus-python.readthedocs.io/en/stable/fields.html#repeated-fields